src/app/payments/helpers.ts

```diff
@@ -145,20 +152,27 @@ export const constructPaymentFlowBuilder = async ({
     usdFromBtcMidPriceFn,
     btcFromUsdMidPriceFn,
   })
-  const builderWithInvoice = uncheckedAmount
-    ? paymentBuilder.withNoAmountInvoice({ invoice, uncheckedAmount })
-    : paymentBuilder.withInvoice(invoice)
+  const builderWithInvoice = invoice
```

Comment:
Not so happy about this 3 way logic in a helpers method. Its not unit tested.
Either we should unify the builder interface and push the logic into the builder or have different entry points ie un-merge the helper function.